src/test/java/org/apache/commons/dbcp2/datasources/TestSharedPoolDataSource.java

```diff
@@ -556,7 +554,7 @@ private void doTestPoolCallableStatements(final PrepareCallCallback callBack)
         conn.close();
         conn = null;
 
-        conn = myDs.getConnection();
+        conn = ((DataSource) spDs).getConnection();
```

Comment:
This is not clearer than what was before. I would not do this.

Reply:
HI @garydgregory 
reverted
TY